foundry/textureLoader.d.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * A Loader class which helps with loading video and image textures
+ */
+declare class TextureLoader {
+  constructor();
+
+  cache: Map<string, PIXI.Texture>;
+
+  /**
+   * Load all the textures which are required for a particular Scene
+   * @param scene - The Scene to load
+   * @returns Promise<void>
+   */
+  static loadSceneTextures(scene: Scene): Promise<void>;
+
+  /**
+   * Load an Array of provided source URL paths
+   * @param sources - The source URLs to load
+   * @param message - The status message to display in the load bar
+   * @returns A Promise which resolves once all textures are loaded
+   */
+  load(sources: string[], { message }?: { message?: string }): Promise<void>;
+
+  /**
+   * Load a single texture on-demand from a given source URL path
+   * @param src -
+   * @returns Promise\<PIXI.Texture\>
+   */
+  loadTexture(src: string): Promise<PIXI.Texture>;
+
+  /**
+   * Log texture loading progress in the console and in the Scene loading bar
+   * @internal
+   */
+  protected _onProgress(src: string, progress: TextureLoader.Progress, message: string): void;
+
+  /**
+   * Log failed texture loading
+   * @internal
```

Comment:
`protected` already marks this as for internal use by the class. We generally use `@internal` for our additions.